src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/TentativeMethodNode.cs

```diff
@@ -87,5 +87,13 @@ public override bool RepresentsIndirectionCell
         public override int ClassCode => 0x562912;
 
         public override bool IsShareable => ((ObjectNode)_methodNode).IsShareable;
+
+        protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFactory factory)
+        {
+            Debug.Assert(factory.Target.Architecture == TargetArchitecture.Wasm32); // only Web Assembly target needs to add the ThrowHelpers.ThrowBodyRemoved dependency like this.
```

Comment:
This assert is going to hit outside WASM because we can ComputeNonRelocation everywhere.

Would be better to do the virtual method I suggested in the comment on AssemblyStubNode.cs.